BusyList/Handlers/HelpHandler.cs

```diff
@@ -0,0 +1,51 @@
+﻿using BusyList.Commands;
+using BusyList.HelpSystem;
+using System;
+
+namespace BusyList.Handlers
+{
+    public class HelpHandler : IHandler<HelpCommand>
+    {
+        private readonly HelpProvider _helpProvider;
+
+        public HelpHandler(HelpProvider helpProvider)
+        {
+            _helpProvider = helpProvider;
+        }
+
+        public void Help()
+        {
+            var helpTexts = _helpProvider.GetAllHelpText();
+
+            foreach (var (name, description) in helpTexts)
+            {
+                Console.WriteLine($"{name}: {description}");
+            }
+
+            Console.WriteLine("For further information type help [keyword] e.g help add");
+        }
+
+        public void Run(HelpCommand command)
+        {
+            if (command.name == null)
+            {
+                Help();
+
+                return;
+            }
+
+            var (description, syntax) = _helpProvider.GetHelpText(command.name);
```

Comment:
Should probably use `.ToLowerInvariant()` here too so we don't get a miss in the dictionary if the user enters ´help Add`.